java/sidecar/src/main/java/com/instaclustr/jersey/JerseyServerModule.java

```diff
@@ -28,17 +51,94 @@ protected void configure() {
     }
 
     @Provides
-    HttpServer provideHttpServer(final ResourceConfig resourceConfig) throws UnknownHostException, URISyntaxException {
-        // TODO: make URI configurable
-        String protocol = "http://";
-        String port = ":4567/";
-        String hostname = InetAddress.getLocalHost().getHostName();
-        if (hostname.equals("127.0.0.1"))
-        {
-            hostname = "localhost";
+    HttpServer provideHttpServer(final ResourceConfig resourceConfig) {
+
+        logger.info("Listening on " + httpServerUri);
+
+        return JdkHttpServerFactory.createHttpServer(httpServerUri, resourceConfig, false);
+    }
+
+    @Singleton
+    @Provides
+    ObjectMapper provideObjectMapper() {
+
+        final SimpleModule module = new SimpleModule("OperationDeserializerModule");
+        module.addDeserializer(Operation.class, new OperationDeserializer());
+
+        final ObjectMapper objectMapper = new ObjectMapper();
+
+        objectMapper.setDefaultPropertyInclusion(JsonInclude.Include.NON_ABSENT);
+        objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        objectMapper.registerModule(module);
+        objectMapper.registerModule(new Jdk8Module());
+        objectMapper.registerModule(new JavaTimeModule());
+
+        return objectMapper;
+    }
+
+    /**
+     * Implementation of feature where overriding of built-in ObjectMapper is happening.
+     * There is autodiscovery of JacksonFeature because it is on the class path from jersey-media-json-jackson
+     * which provides its default ObjectMapper hence if we want to provide our own, this is the standard way how to do that.
+     */
+    public static class MarshallingFeature implements Feature {
+
+        private final ObjectMapper objectMapper;
+
+        @Inject
+        public MarshallingFeature(final ObjectMapper objectMapper) {
+            this.objectMapper = objectMapper;
+        }
+
+        @Override
+        public boolean configure(FeatureContext context) {
+
+            final JacksonJaxbJsonProvider provider = new JacksonJaxbJsonProvider(objectMapper, JacksonJaxbJsonProvider.DEFAULT_ANNOTATIONS);
+
+            context.register(provider);
+            return true;
         }
-        logger.info("Listening on " + protocol + hostname + port);
+    }
+
+    public static class OperationDeserializer extends StdDeserializer<Operation> {
```

Comment:
Let's instead use Jacksons built-in polymorphic serialisation/deserialisation features, namely the `@JsonTypeInfo` and `@JsonSubTypes` annotations (for starters).

See:
- https://stackoverflow.com/questions/30362446/deserialize-json-with-jackson-into-polymorphic-types-a-complete-example-is-giv#30386694
- https://github.com/FasterXML/jackson-docs/wiki/JacksonPolymorphicDeserialization#12-per-class-annotations